File: ZezeJava/ZezeJava/src/main/java/Zeze/Arch/ProviderSession.java
package Zeze.Arch;

import java.util.concurrent.ConcurrentHashMap;
import Zeze.Builtin.Provider.BLoad;
import Zeze.Util.Str;
import Zeze.Util.TimeCounter;
import org.jetbrains.annotations.NotNull;

public class ProviderSession {
	protected volatile @NotNull BLoad load = new BLoad();
	protected final TimeCounter timeCounter = new TimeCounter(5);

	protected int serverId;
	protected long sessionId;
	protected @NotNull String serverLoadIp = "";
	protected long appVersion;
	protected int serverLoadPort;
	protected boolean disableChoice = false;

	/**
	 * 下面维护和本Session相关的订阅Ready状态。在Session关闭时需要取消Ready状态。
	 * 【仅用于ProviderApp】
	 */
	protected final ConcurrentHashMap<String, ConcurrentHashMap<String, ProviderModuleState>> ServiceReadyStates = new ConcurrentHashMap<>();

	public @NotNull String getServerLoadName() {
		return serverLoadIp + '_' + serverLoadPort;
	}

	@Override
	public @NotNull String toString() {
		return '(' + serverLoadIp + ',' + serverLoadPort + ',' + sessionId + ',' + Str.toVersionStr(appVersion) + ')';
	}

	public long getAppMainVersion() {
		return appVersion >>> 48;
	}

	public long getSessionId() {
		return sessionId;
	}

	public int getServerId() {
		return serverId;
	}

	public @NotNull ConcurrentHashMap<String, ProviderModuleState> getOrAddServiceReadyState(@NotNull String serviceName) {
		return ServiceReadyStates.computeIfAbsent(serviceName, __ -> new ConcurrentHashMap<>());
	}

	public boolean isDisableChoice() {
		return disableChoice;
	}

	public void setDisableChoice(boolean value) {
		disableChoice = value;
	}
}

File: Zeze/Arch/ProviderSession.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Zeze.Builtin.Provider;

namespace Zeze.Arch
{
    public class ProviderSession
    {
		public long SessionId { get; }
		private volatile BLoad Load_ = new BLoad();
		public BLoad Load { get { return Load_; } set { Load_ = value; } }
		public string ServerLoadIp { get; set; } = "";
		public int ServerLoadPort { get; set; }
		public string ServerLoadName => ServerLoadIp + ":" + ServerLoadPort;

		public ProviderSession(long sid)
        {
			SessionId = sid;
        }

		/**
		 * 下面维护和本Session相关的订阅Ready状态。在Session关闭时需要取消Ready状态。
		 * 【仅用于ProviderApp】
		 */
		public ConcurrentDictionary<string, ConcurrentDictionary<string, ProviderModuleState>> ServiceReadyStates = new();

		public ConcurrentDictionary<string, ProviderModuleState> GetOrAddServiceReadyState(string serviceName)
		{
			return ServiceReadyStates.GetOrAdd(serviceName, (key) => new());
		}
	}
}
